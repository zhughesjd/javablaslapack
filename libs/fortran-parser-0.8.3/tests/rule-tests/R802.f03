! Test if-construct
!      if-construct  is  if-then-stmt
!                          block
!                          [ else-if-smt 
!                              block ] ...
!                          [ else-stmt 
!                              block ]
!                        end-if-stmt
!
! Also tests rules R803 (if-then-stmt), R804 (else-if-stmt), R805 (else-stmt), 
! R806 (end-if-stmt), and if-construct-name.  Does not test 
! scalar-logical-expr.
if(x .eq. y) then
   x = 1
else if(x .lt. y) then
   x = 0
else
   x = -1
end if

if(x .eq. y) then
   x = 1
else
   x = -1
end if

if(x .eq. y) then
   x = 1
end if

my_if: if(x .eq. y) then
   x = 1
else if(x .lt. y) then my_if
   x = 0
else my_if
   x = -1
end if my_if

end

