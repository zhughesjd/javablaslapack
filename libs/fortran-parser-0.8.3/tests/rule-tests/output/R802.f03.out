*******************************************
args[1]: R802.f03
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R 613:part-ref: id=[@26,461:461='x',<209>,13:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 713:rel-op:
R 613:part-ref: id=[@30,468:468='y',<209>,13:10] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 710:level-3-expr: relOp=[@28,463:466='.eq.',<45>,13:5]
R 803:if-then-stmt: id=null
R 613:part-ref: id=[@38,479:479='x',<209>,14:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@42,483:483='1',<10>,14:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 613:part-ref: id=[@48,493:493='x',<209>,15:8] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 713:rel-op:
R 613:part-ref: id=[@52,500:500='y',<209>,15:15] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 710:level-3-expr: relOp=[@50,495:498='.lt.',<47>,15:10]
R 804:else-if-stmt: id=null
R 613:part-ref: id=[@60,511:511='x',<209>,16:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@64,515:515='0',<10>,16:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 805:else-stmt: id=null
R 613:part-ref: id=[@71,525:525='x',<209>,18:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 709:add-op: addKeyword=[@75,529:529='-',<35>,18:7]
R 406:int-literal-constant: digitString=[@76,530:530='1',<10>,18:8] kindParam=null
R 306:literal-constant:
R 701:primary:
R 705:add-operand: addOp=[@75,529:529='-',<35>,18:7] numAddOps=0
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 806:end-if-stmt: id=null
R 802:if-construct:
R 213:executable-construct:
R 613:part-ref: id=[@85,543:543='x',<209>,21:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 713:rel-op:
R 613:part-ref: id=[@89,550:550='y',<209>,21:10] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 710:level-3-expr: relOp=[@87,545:548='.eq.',<45>,21:5]
R 803:if-then-stmt: id=null
R 613:part-ref: id=[@97,561:561='x',<209>,22:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@101,565:565='1',<10>,22:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 805:else-stmt: id=null
R 613:part-ref: id=[@108,575:575='x',<209>,24:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 709:add-op: addKeyword=[@112,579:579='-',<35>,24:7]
R 406:int-literal-constant: digitString=[@113,580:580='1',<10>,24:8] kindParam=null
R 306:literal-constant:
R 701:primary:
R 705:add-operand: addOp=[@112,579:579='-',<35>,24:7] numAddOps=0
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 806:end-if-stmt: id=null
R 802:if-construct:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@122,593:593='x',<209>,27:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 713:rel-op:
R 613:part-ref: id=[@126,600:600='y',<209>,27:10] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 710:level-3-expr: relOp=[@124,595:598='.eq.',<45>,27:5]
R 803:if-then-stmt: id=null
R 613:part-ref: id=[@134,611:611='x',<209>,28:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@138,615:615='1',<10>,28:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 806:end-if-stmt: id=null
R 802:if-construct:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@150,635:635='x',<209>,31:10] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 713:rel-op:
R 613:part-ref: id=[@154,642:642='y',<209>,31:17] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 710:level-3-expr: relOp=[@152,637:640='.eq.',<45>,31:12]
R 803:if-then-stmt: id=[@145,625:629='my_if',<209>,31:0]
R 613:part-ref: id=[@162,653:653='x',<209>,32:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@166,657:657='1',<10>,32:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 613:part-ref: id=[@172,667:667='x',<209>,33:8] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 713:rel-op:
R 613:part-ref: id=[@176,674:674='y',<209>,33:15] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 710:level-3-expr: relOp=[@174,669:672='.lt.',<47>,33:10]
R 804:else-if-stmt: id=[@181,682:686='my_if',<209>,33:23]
R 613:part-ref: id=[@186,691:691='x',<209>,34:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@190,695:695='0',<10>,34:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 805:else-stmt: id=[@194,702:706='my_if',<209>,35:5]
R 613:part-ref: id=[@199,711:711='x',<209>,36:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 709:add-op: addKeyword=[@203,715:715='-',<35>,36:7]
R 406:int-literal-constant: digitString=[@204,716:716='1',<10>,36:8] kindParam=null
R 306:literal-constant:
R 701:primary:
R 705:add-operand: addOp=[@203,715:715='-',<35>,36:7] numAddOps=0
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 806:end-if-stmt: id=[@210,725:729='my_if',<209>,37:7]
R 802:if-construct:
R 213:executable-construct:
R 209:execution-part-construct:
R 208:execution-part:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=true hasInternalSubprogramPart=false
Parser exiting normally
********************************************
